components/MashForms/FinalDecisionForm.spec.tsx

```diff
@@ -0,0 +1,102 @@
+import { render, fireEvent, screen, waitFor } from '@testing-library/react';
+import { mockedMashReferral } from 'factories/mashReferral';
+import { submitFinalDecision } from 'utils/api/mashReferrals';
+import { mockedWorker } from 'factories/workers';
+import FinalDecisionForm from './FinalDecisionForm';
+
+jest.mock('utils/api/mashReferrals');
+
+const mockPush = jest.fn();
+jest.mock('next/router', () => ({
+  useRouter: () => ({
+    push: mockPush,
+  }),
+}));
+
+describe('#FinalDecisionForm', () => {
+  beforeEach(() => {
+    jest.resetAllMocks();
+
+    render(
+      <FinalDecisionForm
+        referral={mockedMashReferral}
+        workerEmail={mockedWorker.email}
+      />
+    );
+  });
+
+  it('should render correctly', () => {
+    expect(screen.getByText('Document the decision'));
+
+    expect(screen.getByText('Make final decision'));
+    expect(screen.getByText('Select referral category'));
+    expect(screen.getByText('Is this contact urgent?'));
+    expect(screen.getByText('Submit'));
+  });
+
+  it('should dynamically render hint text for emailing a mash manager if decision is urgent', () => {
+    expect(
+      screen.queryByText(
+        'Please email your MASH manager about the urgent case.'
+      )
+    ).toBeNull();
+
+    fireEvent.click(screen.getByText('Yes'));
+
+    expect(
+      screen.getByText('Please email your MASH manager about the urgent case.')
+    );
+  });
+
+  it('should disable submit button when submitFinalDecision Submit button is clicked', () => {
+    expect(screen.getByText('Submit')).not.toHaveAttribute('disabled');
+
+    fireEvent.click(screen.getByText('Submit'));
+
+    expect(screen.getByText('Submit')).toHaveAttribute('disabled');
+  });
+
+  it('should call submitFinalDecision Submit button is clicked', () => {
+    (submitFinalDecision as jest.Mock).mockResolvedValue(true);
+
+    fireEvent.click(screen.getByText('Submit'));
+
+    expect(submitFinalDecision).toBeCalledWith(
+      mockedMashReferral.id,
+      mockedWorker.email,
+      'NFA',
+      'Abuse linked to faith or belief',
+      false
```

Comment:
Where do these values come from in the test?

Reply:
FinalDecisionForm keeps track of the form answers in useState, these answers are the useState defaulted values for our form fields.